node/src/services/blockchain/handler.rs

```diff
@@ -72,19 +222,344 @@ impl ActorEventLoop<BlockchainService> for BlockchainServiceEventLoop {
+    pub fn construct_extrinsic(
+        &self,
+        client: Arc<ParachainClient>,
+        function: impl Into<storage_hub_runtime::RuntimeCall>,
+        nonce: u32,
+    ) -> UncheckedExtrinsic {
+        let function = function.into();
+        let current_block_hash = client.info().best_hash;
+        let current_block = client.info().best_number.saturated_into();
+        let genesis_block = client
+            .hash(0)
+            .expect("Failed to get genesis block hash, always present; qed")
+            .expect("Genesis block hash should never not be on-chain; qed");
+        let period = BlockHashCount::get()
+            .checked_next_power_of_two()
+            .map(|c| c / 2)
+            .unwrap_or(2) as u64;
+        // TODO: Consider tipping the transaction.
+        let tip = 0;
+        let extra: SignedExtra = (
+        frame_system::CheckNonZeroSender::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckSpecVersion::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckTxVersion::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckGenesis::<storage_hub_runtime::Runtime>::new(),
+        frame_system::CheckEra::<storage_hub_runtime::Runtime>::from(generic::Era::mortal(
+            period,
+            current_block,
+        )),
+        frame_system::CheckNonce::<storage_hub_runtime::Runtime>::from(nonce),
+        frame_system::CheckWeight::<storage_hub_runtime::Runtime>::new(),
+        pallet_transaction_payment::ChargeTransactionPayment::<storage_hub_runtime::Runtime>::from(
+            tip,
+        ),
+        cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim::<
+            storage_hub_runtime::Runtime,
+        >::new(),
+    );
+
+        let raw_payload = SignedPayload::from_raw(
+            function.clone(),
+            extra.clone(),
+            (
+                (),
+                storage_hub_runtime::VERSION.spec_version,
+                storage_hub_runtime::VERSION.transaction_version,
+                genesis_block,
+                current_block_hash,
+                (),
+                (),
+                (),
+                (),
+            ),
+        );
+
+        // Getting signer public key.
+        let caller_pub_key = self.keystore.sr25519_public_keys(KEY_TYPE).pop().expect(
+            format!(
+                "There should be at least one sr25519 key in the keystore with key type '{:?}' ; qed",
+                KEY_TYPE
+            )
+            .as_str(),
+        );
+
+        // Sign the payload.
+        let signature = raw_payload
+            .using_encoded(|e| self.keystore.sr25519_sign(KEY_TYPE, &caller_pub_key, e))
+            .expect("The payload is always valid and should be possible to sign; qed")
+            .expect("They key type and public key are valid because we just extracted them from the keystore; qed");
+
+        // Construct the extrinsic.
+        UncheckedExtrinsic::new_signed(
```

Comment:
What are the properties of an `UncheckedExtrinsic`?

Reply:
After a couple of type aliases, it resolves to this: https://github.com/paritytech/polkadot-sdk/blob/3c3d6fceb82372a3019b37117aa453d564b212de/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs#L69